### DISCUSSION TITLE
Repository branches

### DISCUSSION THREAD

**Discussion by majkinetor** [1Remote/1Remote#408] [_Development]
@VShawn, do you think we should have more common branch setup.

I would go with `main` instead of `1Remote_dev` and feature branches as you need them, or for smaller work, push directly to main. We are not in the kind of project that needs to have development branch IMO, so that main can be easily bug fixed without risking a release with half baked feature (this can be done from tag anyway, if it is ever needed, as we will always have tag on release).

On a related note, branches that are not needed any more should be removed.

What do you think ?

**Comment by VShawn** [comment]
I have tried using branches for all modifications, but found it to be very cumbersome, especially when making small changes.

but since `or for smaller work, push directly to main` 

I agree, we can make feature branches for new feature, bug fix and refacing can be push to `main`.

**Reply by majkinetor** [reply]
Oh, yes, I generally do it like this for projects of this type, where most of the work is done by 1 or 2 devs:

1. Push to main all the time, no branches required, even for half work 
     - If half work is expected to last for some time and you still want to do it on main, introduce feature flag.
3. Create feature branch only for '_not sure how it will go, maybe I will abandon this_' or '_it takes a lot of time_' work
4. If you still managed to fuck up the main branch, push revert commits

Nothing beats easiness of working with main. The only thing that is **must** for me is that main should always build without errors (or if not, for it to be fixed during the day).
